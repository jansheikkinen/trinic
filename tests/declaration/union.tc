// Union

union TestUnion
  x: int8, y: float8
end

union TestUnionB(T, V)
  x: T, y: V
end
