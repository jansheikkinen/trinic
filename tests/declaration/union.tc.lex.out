
### LEXER ###
000:        UNION,     (null), (003, 005)
001:   IDENTIFIER,  TestUnion, (003, 015)
002:   IDENTIFIER,          x, (004, 003)
003:            :,     (null), (004, 004)
004:         INT8,     (null), (004, 009)
005:            ,,     (null), (004, 010)
006:   IDENTIFIER,          y, (004, 012)
007:            :,     (null), (004, 013)
008:       FLOAT8,     (null), (004, 020)
009:          END,     (null), (005, 003)
010:        UNION,     (null), (007, 005)
011:   IDENTIFIER, TestUnionB, (007, 016)
012:            (,     (null), (007, 017)
013:   IDENTIFIER,          T, (007, 018)
014:            ,,     (null), (007, 019)
015:   IDENTIFIER,          V, (007, 021)
016:            ),     (null), (007, 022)
017:   IDENTIFIER,          x, (008, 003)
018:            :,     (null), (008, 004)
019:   IDENTIFIER,          T, (008, 006)
020:            ,,     (null), (008, 007)
021:   IDENTIFIER,          y, (008, 009)
022:            :,     (null), (008, 010)
023:   IDENTIFIER,          V, (008, 012)
024:          END,     (null), (009, 003)
